src/components/multi-select/index.tsx

```diff
@@ -0,0 +1,144 @@
+import { Dimensions, ScrollView, StyleSheet, ViewStyle } from 'react-native';
+import type { OptionType, Select } from '@mobile-reality/react-native-select-pro';
+import type {
+    OnPressSelectControlType,
+    OnSetPosition,
+} from '@mobile-reality/react-native-select-pro';
+
+import type { OptionalToRequired } from '../../helpers';
+import { parsePercentageToNumber } from '../../helpers/parsePercentageToNumber';
+import type { State } from '../../state/types';
+import type { DispatchType } from '../../state/types';
+import { SelectInput } from '../select-input';
+
+import { MultiSelectedOption } from './components/selected-option';
+
+type FromSelectComponentProps = Pick<
+    ComponentPropsWithRef<typeof Select>,
+    | 'selectControlTextStyle'
+    | 'placeholderText'
+    | 'selectControlStyle'
+    | 'disabled'
+    | 'searchPattern'
+    | 'searchable'
+    | 'multiSelection'
+>;
+
+type SelectControlProps = OptionalToRequired<
+    FromSelectComponentProps & Pick<State, 'selectedOption'>
+>;
+
+type Props = {
+    onPressRemove: (option: OptionType | null) => void;
+    onPressSelectControl: OnPressSelectControlType;
+} & SelectControlProps &
+    Pick<State, 'isOpened' | 'selectedOption' | 'searchValue'> & {
+        dispatch: DispatchType;
+    } & { setPosition: OnSetPosition };
+
+export const MultiSelect = ({
+    searchable,
+    selectControlTextStyle,
+    selectControlStyle,
+    selectedOption,
+    placeholderText,
+    onPressRemove,
+    disabled,
+    dispatch,
+    isOpened,
+    onPressSelectControl,
+    searchPattern,
+    searchValue,
+    setPosition,
+    multiSelection,
+}: Props) => {
+    const selectedOptionTyped = selectedOption as OptionType[];
+
+    const resolveSelectedOptions = () => {
+        if (!selectedOptionTyped) {
+            if (searchable) {
+                return null;
+            }
+            return (
+                <MultiSelectedOption
+                    isPlaceholder={true}
+                    option={null}
+                    optionWidth={'100%'}
+                    placeholderText={placeholderText}
+                    selectControlTextStyle={selectControlTextStyle}
+                />
+            );
+        }
+
+        const optionWidth = () => {
+            const WIDTH_TRESSHOLD = 100;
+            const WIDTH_OFFSET = 72;
+            const length = selectedOptionTyped.length;
+            const initialWidth = selectControlStyle ? (selectControlStyle as ViewStyle).width : 100;
+            let calculatedWidth = 100;
+            if (typeof initialWidth === 'number') {
+                calculatedWidth = (initialWidth - WIDTH_OFFSET) / length;
+                if (calculatedWidth < WIDTH_TRESSHOLD) {
+                    return WIDTH_TRESSHOLD;
+                }
+                return Math.floor(calculatedWidth);
+            }
+            if (typeof initialWidth === 'string') {
+                const screenWidth = Dimensions.get('window').width;
```

Comment:
https://reactnative.dev/docs/usewindowdimensions is preffered

Reply:
👍